### PR TITLE
Reducing download info dependencies

### PR DESCRIPTION
**Pull request by ouchadam**
Even less behaviour on the file download info!

- extracts the restart time creation

- removes unused allow network file size check

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/165/
Test FAILed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/166/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java [hunk 1]
```diff
@@ -352,7 +354,7 @@ private boolean updateLocked() {
                 isActive = true;
             }
 
-            nextRetryTimeMillis = downloadBatch.nextActionMillis(now, nextRetryTimeMillis);
+            nextRetryTimeMillis = downloadBatch.nextActionMillis(now, nextRetryTimeMillis, restartTimeCreator);
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java]: could be `nextRetryTimeMillis = restartTimer.nextActionMillis();`

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadReadyChecker.java [hunk 2]
```diff
@@ -53,7 +55,7 @@ private boolean isDownloadManagerReadyToDownload(FileDownloadInfo downloadInfo)
             case DownloadStatus.WAITING_TO_RETRY:
                 // download was waiting for a delayed restart
                 final long now = systemFacade.currentTimeMillis();
-                return downloadInfo.restartTime(now) <= now;
+                return restartTimeCreator.restartTime(downloadInfo, now) <= now;
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadReadyChecker.java]: could be `return restartTimer.getRestartTime(downloadInfo);`

File: library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java [hunk 3]
```diff
@@ -237,32 +228,6 @@ private int translateNetworkTypeToApiFlag(int networkType) {
         }
     }
 
-    /**
-     * Check if the download's size prohibits it from running over the current network.
-     *
-     * @return one of the NETWORK_* constants
-     */
-    private NetworkState checkSizeAllowedForNetwork(int networkType) {
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java]: why was this unused though? It's a legit use case https://github.com/novoda/download-manager/blob/master/library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java#L724

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadReadyChecker.java (hunk 2 above)
Comment (ouchadam) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadReadyChecker.java]: this instance returns a boolean

File: library/src/main/java/com/novoda/downloadmanager/lib/RestartTimeCreator.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    private final RandomNumberGenerator randomNumberGenerator;
+
+    public RestartTimeCreator(RandomNumberGenerator randomNumberGenerator) {
+        this.randomNumberGenerator = randomNumberGenerator;
+    }
+
+    /**
+     * Returns the time when a download should be restarted.
+     */
+    public long restartTime(FileDownloadInfo fileDownloadInfo, long now) {
```
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/RestartTimeCreator.java]: relying on an something external to get the current time doesn't seem very reliable, what about passing in a clock?
https://github.com/novoda/all-4/blob/master/core/src/main/java/com/novoda/all4/download/expiry/ExpiryCalculator.java#L34-L38

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadReadyChecker.java (hunk 2 above)
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadReadyChecker.java]: whatever, all I'm saying is the behaviour can now be moved into that class `return restartTimer.shouldRestart(downloadInfo);`

File: library/src/main/java/com/novoda/downloadmanager/lib/RestartTimeCreator.java (hunk 4 above)
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/RestartTimeCreator.java]: could use the `RealSystemFacade` to get the current time https://github.com/novoda/download-manager/blob/master/library/src/main/java/com/novoda/downloadmanager/lib/RealSystemFacade.java

**Pull request closed without merging**